tests/components/coinbase/test_config_flow.py

```diff
@@ -0,0 +1,332 @@
+            result["flow_id"],
+            user_input={
+                CONF_CURRENCIES: [],
+                CONF_EXCHANGE_RATES: [GOOD_EXCHNAGE_RATE],
+            },
+        )
+    assert result2["type"] == "create_entry"
+
+
+async def test_form_bad_exchange_rate(hass):
+    """Test we handle a bad exchange rate."""
+    with patch(
+        "coinbase.wallet.client.Client.get_current_user",
+        return_value=mock_get_current_user(),
+    ), patch(
+        "coinbase.wallet.client.Client.get_accounts", new=mocked_get_accounts
+    ), patch(
+        "coinbase.wallet.client.Client.get_exchange_rates",
+        return_value=mock_get_exchange_rates(),
+    ):
+        config_entry = await init_mock_coinbase(hass)
+        result = await hass.config_entries.options.async_init(config_entry.entry_id)
+        await hass.async_block_till_done()
+        result2 = await hass.config_entries.options.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_CURRENCIES: [],
+                CONF_EXCHANGE_RATES: [BAD_EXCHANGE_RATE],
+            },
+        )
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "exchange_rate_unavaliable"}
+
+
+async def test_option_catch_all_exception(hass):
+    """Test we handle an unknown exception in the option flow."""
+    with patch(
+        "coinbase.wallet.client.Client.get_current_user",
+        return_value=mock_get_current_user(),
+    ), patch(
+        "coinbase.wallet.client.Client.get_accounts", new=mocked_get_accounts
+    ), patch(
+        "coinbase.wallet.client.Client.get_exchange_rates",
+        return_value=mock_get_exchange_rates(),
+    ):
+        config_entry = await init_mock_coinbase(hass)
+        result = await hass.config_entries.options.async_init(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    with patch(
+        "coinbase.wallet.client.Client.get_accounts",
+        side_effect=Exception,
+    ):
+        result2 = await hass.config_entries.options.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_CURRENCIES: [],
+                CONF_EXCHANGE_RATES: ["ETH"],
+            },
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_yaml_import(hass):
+    """Test YAML import works."""
+    conf = {
+        CONF_API_KEY: "123456",
+        CONF_YAML_API_TOKEN: "AbCDeF",
+        CONF_CURRENCIES: ["BTC", "USD"],
+        CONF_EXCHANGE_RATES: ["ATOM", "BTC"],
+    }
+    with patch(
+        "coinbase.wallet.client.Client.get_current_user",
+        return_value=mock_get_current_user(),
+    ), patch(
+        "coinbase.wallet.client.Client.get_accounts", new=mocked_get_accounts
+    ), patch(
+        "coinbase.wallet.client.Client.get_exchange_rates",
+        return_value=mock_get_exchange_rates(),
+    ):
+        result = await hass.config_entries.flow.async_init(
```

Comment:
We normally patch the integration setup functions, like we do in the user step test, on all create entry results to avoid setting up the whole integration.